backend/service/planInvite.ts

```diff
@@ -0,0 +1,70 @@
+import { type User, InviteStatus, Role } from '@prisma/client';
+import PlanMembershipModel from '../models/planMembership';
+import PlanInviteModel from '../models/planInvite';
+
+export default class PlanInviteService {
+  static async setUpInviteePlanMembership(planInviteID: string, user: User): Promise<string> {
+    const planInviteData = await PlanInviteModel.readOnePlanInvite({ id: planInviteID })
+
+    if (planInviteData === null) throw new Error('fail to fetch planID from planInvite model');
+
+    // create plan membership
+    await PlanMembershipModel.dbCreateOnePlanMembership({
+      planID: planInviteData.planID,
+      userID: user.id,
+      inviteStatus: InviteStatus.ACCEPTED,
+      role: Role.FRIEND
+    })
+
+    // return plan id
+    return planInviteData.planID
+  }
+
+  static async isUserAlreadyPlanMember(planInviteID: string, userID: string) {
+    const planInviteData = await PlanInviteModel.readOnePlanInvite({
+      id: planInviteID
+    })
+
+    if (planInviteData === null) throw new Error('unable to find plan')
+
+    const isUserInPlan = await PlanMembershipModel.dbReadOnePlanMembership({ planID: planInviteData.planID, userID })
+
+    // if no record is found, then user is not in plan
+    if (isUserInPlan === null) return false
+
+    return true
+  }
+
+  static async isUserEmailMatchPlanInviteEmail(inviteeEmail: string, planInviteID: string): Promise<boolean | null> {
+    try {
+      // search the PlanInvite model to see if there is a record of the inviteeEmail and planInviteID
+      const res = await PlanInviteModel.readOnePlanInvite({
+        inviteeEmail,
+        id: planInviteID
+      })
+
+      // if no record is found, then email does not match
+      if (res === null) return false
+
+      return true
```

Comment:
kind of the same here, we could just return `!!res` provided the readOnePLanInvite returns undefined/null if it cannot find .

Also REALLY minor point, but technically instead of doing a readOne (which returns a full document), you could just return the count instead

https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#count

Reply:
gotcha